### PR TITLE
feat: completes the functions with conditional statements and loops f…

### PR DESCRIPTION
**Pull request by RichHawkins**
…or lesson_06

### CODE REVIEW

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java [hunk 1]
```diff
@@ -21,7 +21,7 @@ public static void main(String[] args) {
    * @return True if the age corresponds to a voting age and false otherwise.
    */
   public static boolean canVote(int age) {
-    return false;
+    return age >= 18 ? true : false;
```
Comment (anthonydmays) [inline lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:24]: Don't need the ternary part, will still work.

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java [hunk 2]
```diff
@@ -61,7 +89,11 @@ public static String convertGpaToLetterGrade(double gpa) {
    * @return The factorial of n.
    */
   public static int computeFactorial(int n) {
-    return 0;
+    int factorial = 1;
```
Comment (anthonydmays) [inline lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:92]: Should throw an exception if `n < 0`, think you can make that happen?

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java [hunk 3]
```diff
@@ -81,7 +117,17 @@ public static double addNumbers(double[] values) {
    * @return An array containing the first `n` fibonacci values.
    */
   public static int[] getFirstNFibonacciNumbers(int n) {
-    return null;
+    int[] fibNums = new int[n];
+    int num1 = 1;
+    int num2 = 1;
+    for (int i = 0; i < n; i++) {
+      fibNums[i] = num1;
```
Comment (anthonydmays) [inline lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:124]: Would be more naturally expressed as `fibNums[i] = fibNums[i - 1] + fibNums[i - 2]`, then special case values of `n < 2`.

**Review by anthonydmays — CHANGES_REQUESTED** [review]

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java (hunk 1 above)
Comment (RichHawkins) [reply lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:24]: removed ternary part!

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java (hunk 2 above)
Comment (RichHawkins) [reply lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:92]: This should now throw an exception if n < 0.

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java (hunk 3 above)
Comment (RichHawkins) [reply lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:124]: Implemented more natural expression and special cases for values less than 2.

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java [hunk 4]
```diff
@@ -71,7 +107,11 @@ public static int computeFactorial(int n) {
    * @return The sum of all the values.
    */
   public static double addNumbers(double[] values) {
-    return 0;
+    double sum = 0;
```
Comment (anthonydmays) [inline lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:110]: You can do this is one line of code using Stream.reduce. See if you can make that work.

**Review by anthonydmays — APPROVED** [review]
LGTM

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java (hunk 4 above)
Comment (RichHawkins) [reply lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java]: Cool! Reduced to one line!

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java [hunk 5]
```diff
@@ -71,7 +108,8 @@ public static int computeFactorial(int n) {
    * @return The sum of all the values.
    */
   public static double addNumbers(double[] values) {
-    return 0;
+
+    return Arrays.stream(values).reduce(0, (a, b) -> a + b);
```
Comment (anthonydmays) [inline lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:112]: Excellent!

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java [hunk 6]
```diff
@@ -61,7 +90,15 @@ public static String convertGpaToLetterGrade(double gpa) {
    * @return The factorial of n.
    */
   public static int computeFactorial(int n) {
```
Comment (anthonydmays) [inline lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:92]: Looks good! I'd be curious to know if you can implement this recursively. Give it a shot.

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java [hunk 7]
```diff
@@ -103,10 +153,17 @@ public static int binarySearch(int[] values, int start, int end, int value) {
 
     // TODO(you): Finish implementing this algorithm
 
+    if (values[pivotIndex] == value) {
+      return pivotIndex;
+    } else if (values[pivotIndex] > value) {
+      return binarySearch(values, start, pivotIndex - 1, value);
```
Comment (anthonydmays) [inline lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:159]: If you're up for it, here's another challenge: implement this without recursion (aka the iterative approach).

**Review by anthonydmays — DISMISSED** [review]
LGTM

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java (hunk 6 above)
Comment (RichHawkins) [reply lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:92]: Got this working recursively....definitely was a little mind bending but it's pretty cool how it works.

Comment (anthonydmays) [reply lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:92]: Yessir! And excellent work with this.

nit: Please resolve the conversation when you're done with it. Once all conversations are resolved, you should be able to send back to me and I'll see your PR as ready to review.

**Review by anthonydmays — APPROVED** [review]
LGTM

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java (hunk 7 above)
Comment (RichHawkins) [reply lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java]: This was tricky! definitely helping wrap my head around the concepts though.

**Review by anthonydmays — APPROVED** [review]
LGTM

**Pull request closed without merging**